### ISSUE TITLE
Add a gallery with exemplary meson.build projects

### ISSUE THREAD

**Issue by HDembinski** [mesonbuild/meson#11238]
tl;dr: I suggest to a "gallery" of outstanding meson.build projects, as examples to steal code from. 

This would be a page in the documentation which links to exemplary projects that use meson in real-life for non-trivial tasks, for example, building and distributing a Python extension with meson that works on all major platforms. I specifically like the scipy meson.build files, they should be listed there.

Context:

Hi, I am the author of Boost.Histogram and several scientific Python/C++ libraries, I contributed to scipy and matplotlib.

I tried out meson yesterday for the first time and ported one of my projects from CMake to meson. I want to use meson-python, the PEP-517 compliant build backend for Python packages.

The meson documentation is good, but it nevertheless took me quite a few hours for what I thought is a simple project to port. I think the documentation should also include more complex examples and not only the most trivial ones. Eventually, I found most of what I needed in the scipy meson.build files, which are very high quality and deal with the complex issues that one encounters in real life.

**Comment by eli-schwartz** [comment]
https://mesonbuild.com/Users.html exists, but in no particular order.

> exemplary projects that use meson in real-life for non-trivial tasks

The majority of those are surely non-trivial, but haven't been vetted to determine whether they qualify as "exemplary". SciPy is* listed there though. :)

**Comment by HDembinski** [comment]
I think you understand my point already, but let me say that

https://mesonbuild.com/Users.html

is not a substitute for what I am suggesting. A users page just conveys to me that this project is production ready, since I can find many well-known packages there, including Scipy. However, when looking for exemplary meson projects to learn from, this is not where I would naturally go.

I am more think of a page with a structure like this

Exemplerary projects

This page lists exemplary projects that solve comlex real-life problems with meson.

- Build C/C++/Fortran Python extensions and distribute it:
A B C

- Other complex task
D E F

**Comment by tristan957** [comment]
I personally think the current `Users.html` page is not really that good. I think it would be better to only have a curated list of projects, like systemd, postgres, GNOME Builder, etc.

> Build C/C++/Fortran Python extensions and distribute it:

It would definitely be good to call stuff like this out. Makes it easy for a new user to copy paste. Alternatively, the mesonbuild org could host some [template repositories](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) for this kind of thing.

1. `Template for Java with JNI`
2. `Template for C Python extensions`
3. `Template for Fortran Python extensions`

**Comment by dcbaker** [comment]
We have a `meson init` command, adding those kind of templates seems like it's in scope

**Comment by tristan957** [comment]
Never realized that.

**Comment by bonzini** [comment]
Closing - Users.md has been reduced to a smaller number of flagship projects, and between that and `meson init` this can be considered done.
